crates/core/src/operations/load_cdf.rs

```diff
@@ -139,8 +151,13 @@ impl CdfLoadBuilder {
                 .log_store
                 .read_commit_entry(version)
                 .await?
-                .ok_or(DeltaTableError::InvalidVersion(version))?;
-            let version_actions = get_actions(version, snapshot_bytes).await?;
+                .ok_or(DeltaTableError::InvalidVersion(version));
+
+            if snapshot_bytes.is_err() && version >= end && self.enable_out_of_range {
```

Comment:
Do you need this? You adjusted the end version above, so it should never exceed the last version right?

Reply:
From my checks if the start that you provide to `self.log_store.get_latest_version(start)` is greater than the last commit, then it returns that same start, so version could exceed the end of log.

I think I could adjust the call to do  `self.log_store.get_latest_version(0)` so it always start at the beginning and make sure I get the real end, but did not want to change the code too much.

Reply:
I have made this change, use ` self.log_store.get_latest_version(0)` to read the actual last version and use it in the check